### PR TITLE
RSDK-7923 - fix WebRTC connections between a viam-server and a module can involve non loopback interfaces

### PR DESCRIPTION
**Pull request by nicksanford**
[Ticket](https://viam.atlassian.net/browse/RSDK-7923)

Original intent when we added WebRTC connection support between a module & a viam-server is to support sending webrtc track data from the module to the viam-server. We assumed the loopback address would always be used. Testing found that frequently one or the other peer bind to a non loopback interface. This change makes it so that the viam-server & module will both only allow candidates from the other which are on the loopback interface.

Manual Testing:
- [x] new go module, new viam-server: both peers always connect on 127.0.0.1 ports
- [x] old go module (erh/viamrtsp 0.0.3 using rdk version v0.26.0-rc0.0.20240503203304-30f601249ccf), new viam-server: go module might connect on non 127.0.0.1 port, aka no behavior change
- [x] new go module, old viam-server  (v0.28.1): go module might connect on non 127.0.0.1 port, aka no behavior change

### CODE REVIEW

**Comment by Otterverse** [comment]
Adding a label to test a CI issue. (This was just a convienent PR.) Sorry for the noise.

File: services/sensors/client.go [hunk 1]
```diff
@@ -1,6 +1,6 @@
 // Package sensors contains a gRPC based sensors service client
 //
-//nolint:staticcheck
```
Comment (nicksanford) [inline services/sensors/client.go:3]: Done by linter

File: services/sensors/server.go [hunk 2]
```diff
@@ -1,6 +1,6 @@
 // Package sensors contains a gRPC based sensors service server
 //
-//nolint:staticcheck
```
Comment (nicksanford) [inline services/sensors/server.go:3]: Done by linter

File: services/sensors/server_test.go [hunk 3]
```diff
@@ -32,7 +32,7 @@ func TestServerGetSensors(t *testing.T) {
 		sMap := map[resource.Name]sensors.Service{}
 		server, err := newServer(sMap)
 		test.That(t, err, test.ShouldBeNil)
-		//nolint:staticcheck
```
Comment (nicksanford) [inline services/sensors/server_test.go:35]: Done by linter

File: grpc/shared_conn.go [hunk 4]
```diff
@@ -319,3 +325,107 @@ func (sc *SharedConn) Close() error {
...
+	if err != nil {
+		logger.Debugw("Unable to create optional peer connection for module. Skipping WebRTC for module...", "err", err)
+		return nil, err
+	}
+	return pc, nil
+}
+
+// WebRTCLoggerFactory wraps a golog.Logger for use with pion's webrtc logging system.
+type WebRTCLoggerFactory struct {
+	Logger golog.Logger
```
Comment (dgottlieb) [inline grpc/shared_conn.go:377]: I'm guessing this was just copied from goutils for expediency? That's fine if so. I can dabble in figuring out what our longer-term options are here.

**Review by dgottlieb — APPROVED** [review]

File: grpc/shared_conn.go (hunk 4 above)
Comment (nicksanford) [reply grpc/shared_conn.go:377]: yep correct

**Pull request merged**